### ISSUE TITLE
Incorrect scaladoc links to JDK apidocs

### ISSUE THREAD

**Issue by durban** [typelevel/sbt-typelevel#592]
Links to JDK APIs in the generated scaladoc seem to be incorrect. Two examples:

- In the [Cats Effect](https://typelevel.org/cats-effect/api/3.x/cats/effect/std/SecureRandom$.html#javaSecuritySecureRandom[F[_]](implicitevidence$17:cats.effect.kernel.Sync[F]):F[cats.effect.std.SecureRandom[F]]) documentation: `java.security.SecureRandom` links to https://docs.oracle.com/en/java/javase/11/docs/api/java/security/SecureRandom.html (which is a dead link).
- [Here](https://www.javadoc.io/static/dev.tauri/choam-docs_2.13/0.4.0-M2/dev/tauri/choam/refs/Ref.html): `java.util.concurrent.atomic.AtomicReference` links to https://docs.oracle.com/en/java/javase/17/docs/api/java/util/concurrent/atomic/AtomicReference.html (which is also a dead link).

In theory this can be an sbt/scaladoc bug, but I see some settings [here](https://github.com/typelevel/sbt-typelevel/blob/main/settings/src/main/scala/org/typelevel/sbt/TypelevelSettingsPlugin.scala#L368), which seem related, but somehow they're not applied(?).

**Comment by armanbilge** [comment]
Hmm, maybe we forgot to prefix it with `Compile /`? 🤔 
https://github.com/typelevel/sbt-typelevel/blob/1f6850a0906ffdb47086a24d268e3f51b9a43408/settings/src/main/scala/org/typelevel/sbt/TypelevelSettingsPlugin.scala#L370

**Comment by armanbilge** [comment]
No, that seems fine. So now I suspect maybe we are not mapping the right thing 🤔 

https://github.com/typelevel/sbt-typelevel/blob/1f6850a0906ffdb47086a24d268e3f51b9a43408/settings/src/main/scala/org/typelevel/sbt/TypelevelSettingsPlugin.scala#L388

**Comment by armanbilge** [comment]
I think we should try refreshing our cargo-cult with exactly whats in Scala.js, for example this part seems interesting.

https://github.com/scala-js/scala-js/blob/e0afd0f096c6b096ce7ad17f46c07142ac80c7ef/project/Build.scala#L428-L430

**Comment by armanbilge** [comment]
Tried fixing this in 4b8962bdb2921a578e9cb66cf89515bf5ecdf09d but it didn't fix Choam 😕 so either we're still not doing it right, or perhaps sbt is overriding us.

**Comment by durban** [comment]
Thanks for trying! This whole apiMappings setting is very mysterious to me, so I don't have any ideas...

**Comment by durban** [comment]
Is this thing working _anywhere_? I see that in the scala-js API it seems to work; I suspect that's because it uses the old (JVM 8) url...

- looking at the code, sbt generates `-doc-external-doc` cmdline options for scaladoc: https://github.com/sbt/sbt/blob/1.9.x/main/src/main/scala/sbt/Opts.scala#L41
- scaladoc however, seems to _ignore_ that option for JDK symbols: https://github.com/scala/scala/blob/2.13.x/src/scaladoc/scala/tools/nsc/doc/model/MemberLookup.scala#L64C1-L64C1
- and instead it uses it's own URL, which seems to be incorrect(?): https://github.com/scala/scala/blob/2.13.x/src/scaladoc/scala/tools/nsc/doc/model/MemberLookup.scala#L123

**Comment by durban** [comment]
There is a strange interaction with the `-release` option, but it doesn't seem `sbt-typelevel` specific, so I've opened https://github.com/scala/bug/issues/12820.

**Comment by armanbilge** [comment]
Aha, I see, thanks for investigating. But this does illustrate our setting is not really doing anything.
